samcli/commands/build/click_container.py

```diff
@@ -0,0 +1,27 @@
+"""
+Module to check container based cli parameters
+"""
+import click
+
+
+class ContainerOptions(click.Option):
+    """
+    Preprocessing checks for presence of --use-container flag for container based options.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.require_container: bool = kwargs.pop("require_container", False)
+
+        super().__init__(*args, **kwargs)
+
+    def handle_parse_result(self, ctx, opts, args):
+        if self.require_container:
+            if "use_container" not in opts and opts.get(self.name) is not None:
+                msg = f"""\
+Missing required parameter, with --{self.name.replace("_", "-")} set.
+
+Must provide the --use-container flag in order to use --{self.name.replace("_", "-")} flag.\
+                """
+                raise click.UsageError(msg)
+            self.prompt = None
```

Comment:
what does this `self.prompt = None` do?

Reply:
this is an attribute inherited from click.Options

Reply:
why we set prompt to `None` here, the intention is not clear here, maybe a comment would help.

Reply:
Because click options have a prompt field which can be set to generate user input prompts, this is just to make sure no prompting happens

Reply:
exactly, it totally make sense after you explain it. It is not clear just by looking at the code, that's why I think a comment here would greatly help.